src/core/documentation/documentation.ts

```diff
@@ -0,0 +1,158 @@
+import { Model } from "../model/model";
+import { Element } from "../model/element";
+import * as assert from "assert";
+import { Section } from "./section";
+import { Decision } from "./decision";
+import { SoftwareSystem } from "../model/softwareSystem";
+import { DecisionStatus } from "./decisionStatus";
+import { Format } from "./format";
+
+export class Documentation {
+    private model: Model;
+    public sections: Set<Section> = new Set();
+    public decisions: Set<Decision> = new Set();
+
+    constructor(model: Model) {
+        this.model = model;
+    }
+
+    public addSection(
+        element: Element,
```

Comment:
the element parameter is optional - using `element?: Element` won't compile though without moving the parameter to the end. I'd suppose to use `element: Element | undefined`

Reply:
this will then also require changing the call to this function from the test workspace to use `undefined` instead of  `null`